sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -94,6 +101,84 @@ def on_request(self, request):
         self._update_headers(request.http_request.headers, self._token.token)
 
 
+class ChallengeAuthenticationPolicy(HTTPPolicy):
+    """Base class for policies that authorize requests with bearer tokens and expect authentication challenges
+
+    :param ~azure.core.credentials.TokenCredential credential: an object which can provide access tokens, such as a
+        credential from :mod:`azure.identity`
+    :param str scopes: required authentication scopes
+    """
+
+    def __init__(self, credential, *scopes, **kwargs):  # pylint:disable=unused-argument
+        # type: (TokenCredential, *str, **Any) -> None
+        super(ChallengeAuthenticationPolicy, self).__init__()
+        self._scopes = scopes
+        self._credential = credential
+        self._token = None  # type: Optional[AccessToken]
+
+    def _need_new_token(self):
+        # type: () -> bool
+        return not self._token or self._token.expires_on - time.time() < 300
```

Comment:
Do we want to make '300' configurable?

Reply:
Not initially. We could add configuration later.